contributor-book/philosophy_0_80.md

```diff
@@ -0,0 +1,182 @@
+
+### Scripting language
+
+The intent of Nushell's design is to be a scripting language. Scripting can take many forms, from simple pipelines to large scripts. Nushell should handle these with ease by meeting reasonable expectations in terms of:
+- Modularity
+- Readability
+- Programming language features and convenience\
+
+### Structured data
+
+In Nushell, all data is "structured". For Nushell, this means that values can take shapes beyond just a simple string.
+
+Values in Nushell can be records, lists, tables, binary data, and more. Being able to convert into structured data and work with structured data is fundamental to Nushell.
+
+### Pipelines
+
+Nushell takes the Unix philosophy of pipelines to heart. Commands should be built with the intent of composition. Nushell leverages this composition via the use of pipes (`|`), just like Unix pipelines.
+
+Composing commands, both built-in and user-defined, is a core piece of Nushell. The design of Nushell and its standard library must support both easily composing commands as well as allowing the user to easily create compose-able commands.
+
+## Command signatures and their parts
+
+### Signature
+
+The signature of the command describes the following:
+
+- The name of the command
+- The usage information (documentation) for the command
+- The name and type of:
+    - Named parameters
+    - Positional parameters
+- The type of:
+    - Input
+    - Output
+
+### When to use input
+
+Input is intended for one or both of:
+- Pipeline composition
+- Streams of data
+
+### When to use output
+
+All return values are output.
+
+### When to use positional parameters
+
+Use a positional parameter when:
+- A parameter is required
+
+### When to use rest parameters
+
+Use a rest parameter when:
+- A command takes some number of additional optional arguments of the same type
+
+### When to use named parameters
+
+Use a named parameter (a flag parameter) when:
+- The value you need is optional
+
+### When to use a switch flag
+
+Use a switch flag when:
+- You need to allow the user to change the default action of the command
+
+### Comments
+
+Commands should be commented to describe their function and usage. This documentation should also cover parameters and their use.
+
+## Core categories
+
+The core language and standard library needs to cover the following categories to support common use cases for Nushell:
+
+* Filesytem
+* Operating system
+* Manipulating the environment
+* Parsing string data into structure data
+* Querying, filtering, and manipulating structured data
+* Network connectivity
+    * note: Network support is fundamental because with it users can easily acquire and install Nushell extensions
+* Basic formats:
+    * CSV and JSON support
```

Comment:
I think a case could be made for keeping TOML/YAML/XML in the core as well. But I'd be open to just CSV+JSON.

Reply:
+1 for TOML

Reply:
I think we can have more format supported, in additional to @rgwood mentioned, I'd like to have `ssv`, `ini`, 'xlsx` supports.   But these would introduce too much dependencies into nushell core itself, I'm not sure...

Reply:
I changed it so the exact formats are "to be decided" rather than listing them now. We should look to see which ones are used a lot in practice in current Nushell.